### PR TITLE
manifest: hal_nxp: Pull in change to fix irqsteer mask computation

### LINKED ISSUE #80681: Fix irqsteer mask computation
**Describe the bug**

For IRQ_STEER, lower CHn_MASK register indexes are used
to mask higher interrupt IDs. For instance, in the case
of i.MX8MP, the mapping is as follows:

	CHn_MASK4 => masks interrupts [31:0]
	CHn_MASK3 => masks interrupts [63:32]
	CHn_MASK2 => masks interrupts [95:64]
	CHn_MASK1 => masks interrupts [127:96]
	CHn_MASK0 => masks interrupts [159:128]

The `IRQSTEER_GetRegIndex()` function is used to fetch
the CHn_MASK register index based on a given slice. The term
"slice" is used to refer to an index of a CHn_MASK register in
the set of CHn_MASK registers assigned to a certain master. Assuming
the following partition scheme (i.MX8MP):

	{ CHn_MASK4 } is assigned to MASTER0
	{ CHn_MASK3, CHn_MASK2 } is assigned to MASTER1
	{ CHn_MASK1, CHn_MASK0 } is assiged to MASTER2

CHn_MASK3 would be at slice (index) 0, CHn_MASK2 would be at
slice 1, CHn_MASK1 would be at slice 0 and so on.

To compute the CHn_MASK register index found at a given slice,
`IRQSTEER_GetRegIndex()` uses a base index, which is either
the lowest or the highest CHn_MASK register index in a master's
partition (for instance, for MASTER1 the higher would be 3 and
the lowest would be 2).

For IRQ_STEER instances with an uneven number of CHn_MASK registers
the base is the lowest index, while in the case of instances with
an even number of CHn_MASK registers it's exactly the opposite. This
is an issue because the software using this function might expect
the base to either be the lowest or the highest index ALL THE TIME
(since this affects the order in which the CHn_MASK register indexes
are returned).

- Platform HIFI4 DSP on i.MX8MP
- This is not a regression, the bug was introduced on i.MX8MP from the start.

**To Reproduce**

Build Sound Open Firmware
1. cd sof
2. ./scripts/xtensa-build-zephyr.py -p -m imx8mp

**Expected behavior**

IRQsteer should deliver the correct interrupts.

**Impact**

No driver using interrupts can work without this fix.

**Environment (please complete the following information):**

 - Linux LTS 6.6
 - Toolchain zephyr-sdk-0.16.5-1

### PR DESCRIPTION
**Pull request by dbaluta**
This fixes irq_steer channel mask index computation for i.MX8MP platform.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/80681

### CODE REVIEW

**Review by LaurentiuM1234 — DISMISSED** [review]
good to go once HAL stuff gets merged

**Comment by dleach02** [comment]
rebase HAL PR and this one

**Comment by dbaluta** [comment]
-- rebased on latest main

**Comment by dbaluta** [comment]
@mmahadevan108  @dleach02  can we please have this bugfix in 4.0?

**Comment by dbaluta** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/80681

**Review by dleach02 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**
